tests/test_client.py

```diff
@@ -143,3 +153,42 @@ def test_bad_endpoints(self):
                     "account": "",
                 },
             )
+
+
+@patch("pasqal_cloud.client.Auth0TokenProvider", FakeAuth0AuthenticationSuccess)
+class TestSDKAuth0AndEndpointsDicts(TestSDKCommonAttributes):
+    @pytest.mark.parametrize(
+        "env, core_endpoint_expected",
+        [
+            ("prod", "https://apis.pasqal.cloud/core-fast"),
+            ("preprod", "https://apis.preprod.pasqal.cloud/core-fast"),
+            ("dev", "https://apis.dev.pasqal.cloud/core-fast"),
+        ],
+    )
+    def test_existing_dict_keys(self, env, core_endpoint_expected):
+        sdk = SDK(
+            project_id=self.project_id,
+            username=self.username,
+            password=self.password,
+            auth0=AUTH0_CONFIG[env],
+            endpoints=PASQAL_ENDPOINTS[env],
+        )
+        assert sdk._client.endpoints.core == core_endpoint_expected
+
+    def test_init_sdk_with_unknown_auth0_dict_key(self):
```

Comment:
I respect the effort but this is just testing that python raises a KeyError when trying to access a key that doesn't exist in a dictionary, I don't think we should test that :wink: